### DISCUSSION TITLE
Feature Request: Add support for loading Amazon Ion files (similarly to the JSON extension)

### DISCUSSION THREAD

**Discussion by anna-geller** [duckdb/duckdb#10940] [Ideas]
Similar to how DuckDB supports the [JSON extension](https://duckdb.org/docs/extensions/json), it would also be great to cover [Amazon Ion](https://amazon-ion.github.io/ion-docs/).

We heavily rely on Ion's [richly-typed](https://amazon-ion.github.io/ion-docs/guides/why.html#rich-type-system) data serialization format, and it would be extremely helpful if we could read and write Ion files in DuckDB.

**Comment by tchiotludo** [comment]
We wanted at first to hire a freelancer to start working on this extension, but @hmb1 raise a warning: 
> Have just download and built the amazon-ion/ion-c repo.
Upon intial inspection it looks like a lot of the work will on integration rather than low level coding.
Mark R is the principal developer of DuckDB so we need to come up with a credible implementation plan to pass by him first

@Mytherin any opinion on this subject.

**Reply by Mytherin** [reply]
I would recommend building an [external extension](https://github.com/duckdb/extension-template), that way you don't need to pass anything by me
